pkg/model/fips.go

```diff
@@ -91,6 +95,15 @@ func EnforceCompliance(ctx *tls.CommonTlsContext) {
 		// Default (unset) is P-256
 		ctx.TlsParams.EcdhCurves = nil
 		return
+	case common_features.PQC:
+		if ctx.TlsParams == nil {
+			ctx.TlsParams = &tls.TlsParameters{}
+		}
+		ctx.TlsParams.TlsMinimumProtocolVersion = tls.TlsParameters_TLSv1_3
+		ctx.TlsParams.TlsMaximumProtocolVersion = tls.TlsParameters_TLSv1_3
+		ctx.TlsParams.CipherSuites = nil
```

Comment:
wouldn't we need a `map` of legal ciphersuites like in the FIPS implementation above? We can't use any that e.g. use RSA instead of ECDSA, right?

Reply:
Envoy docs says that cipher suites are not ignored for TLS 1.3: "If specified, the TLS listener will only support the specified [cipher list](https://commondatastorage.googleapis.com/chromium-boringssl-docs/ssl.h.html#Cipher-suite-configuration) when negotiating TLS 1.0-1.2 (this setting has no effect when negotiating TLS 1.3)."

Reply:
So how do we enforce usage of ECDSA? If an RSA cipher is selected, the curve setting won't apply and this won't be PQC

Reply:
> > OpenSSL does allow setting ciphersuites for TLS v1.3 (see [envoyproxy/envoy#19548  
> > So how do we enforce usage of ECDSA?
> 
> We don't need to care about digital signature. It's decoupled from cipher suites in TLS 1.3. Here is the list of cipher suites for 1.3: https://docs.openssl.org/1.1.1/man1/ciphers/#tls-v13-cipher-suites. Ztunnel has hardcoded `TLS_AES_128_GCM_SHA256` and `TLS_AES_256_GCM_SHA384`, so I think we should use these two, but I am not sure how to do this only for openssl...

I meant for the handshake. But I just read up on it and TLSv1.3 dropped RSA from the handshake, so we should be good.

Reply:
@dgn I added cipher suites.